plugins/tours/tests/phpunit/integration/registerTheTemplateFiles.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Tests for register_the_template_files().
+ *
+ * @package     spiralWebDb\CornerstoneTours\Tests\Integration
+ * @since       1.0.0
+ * @author      Robert Gadon <rgadon107>
+ * @link        https://github.com/rgadon107/cornerstone
+ * @license     GNU-2.0+
+ */
+
+namespace spiralWebDb\CornerstoneTours\Tests\Integration;
+
+use spiralWebDb\Cornerstone\Tests\Integration\Test_Case;
+use function has_filter;
+use function spiralWebDb\CornerstoneTours\_get_plugin_directory;
+use function spiralWebDb\CornerstoneTours\register_the_template_files;
+
+/**
+ * Class Tests_RegisterTheTemplateFiles
+ *
+ * @package spiralWebDb\CornerstoneTours\Tests\Integration
+ * @group   tours
+ * @group   admin
+ */
+class Tests_RegisterTheTemplateFiles extends Test_Case {
+
+	/*
+	 * Test register_the_template_files() is registered to filter 'register_templates_with_template_loader' when event fires.
+	 */
+	public function test_callback_is_registered_to_filter_hook_when_event_fires() {
+		$this->assertTrue( has_filter( 'register_templates_with_template_loader' ) );
+		$this->assertEquals( 10, has_filter( 'register_templates_with_template_loader', 'spiralWebDb\CornerstoneTours\register_the_template_files' ) );
+	}
+
+	/*
+     * Test register_the_template_files() should return template files when given valid path to plugin config.
+     */
+	public function test_should_return_template_files_given_valid_path_to_plugin_config() {
+		$templates = [];
+
+		$this->assertArrayHasKey( 'single', register_the_template_files( (array) $templates ) );
+		$this->assertArraySubset( [ 'single' => [ 'tours' => _get_plugin_directory() . '/src/template/single-tours.php' ] ], register_the_template_files( (array) $templates ) );
+		$this->assertArrayHasKey( 'post_type_archive', register_the_template_files( (array) $templates ) );
+		$this->assertArraySubset( [ 'post_type_archive' => [ 'tours' => _get_plugin_directory() . '/src/template/archive-tours.php' ] ], register_the_template_files( (array) $templates ) );
+	}
```

Comment:
This function merges a given array of template files with the configuration. You want to test:

1. With an empty array of templates, it passes back the configuration.
2. When given an array of templates, it passes back a merged array of templates.

How?

```php
$config = require _get_plugin_directory() . '/config/templates.php';

$this->assertSame(
	$config,
	register_the_template_files( [] )
);

$templates = [
	'single' => [
		'baz' => __DIR__ . '/baz/templates.php',
	],
];
$this->assertSame(
	array_merge_recursive( $templates, $config ),
	register_the_template_files( $templates ) );
}
```

Reply:
@hellofromtonya I haven't worked much with the PHP method `array_merge_recursive()`. I admit that it threw me. I used `var_dump()` inside the test methods to inspect what the function under test and `array_merge_recursive()` returned. Now I understand how and why you set up the test scenarios the way that you did. 

Fixed.